src/lib.rs

```diff
@@ -244,40 +244,72 @@ mod ios {
 ))]
 mod unix {
     use std::{
-        ffi::OsStr,
-        io::{Error, Result},
+        env,
+        ffi::{OsStr, OsString},
+        io,
+        path::{Path, PathBuf},
         process::{Command, ExitStatus, Stdio},
     };
 
-    use which::which;
+    pub fn that<T: AsRef<OsStr> + Sized>(path: T) -> io::Result<ExitStatus> {
+        let path = path.as_ref();
+        let open_handlers = [
+            ("xdg-open", &[path] as &[_]),
+            ("gio", &[OsStr::new("open"), path]),
+            ("gnome-open", &[path]),
+            ("kde-open", &[path]),
+            ("wslview", &[&wsl_path(path)]),
+        ];
 
-    pub fn that<T: AsRef<OsStr> + Sized>(path: T) -> Result<ExitStatus> {
-        ["xdg-open", "gio", "gnome-open", "kde-open", "wslview"] // Open handlers
-            .iter()
-            .find(|it| which(it).is_ok()) // find the first handler that exists
-            .ok_or(Error::from_raw_os_error(0)) // If not found, return err
-            .and_then(|program| {
-                // If found run the handler
-                if *program == "gio" {
-                    Command::new(program)
-                        .stdout(Stdio::null())
-                        .stderr(Stdio::null())
-                        .arg("open")
-                        .arg(path.as_ref())
-                        .spawn()?
-                        .wait()
-                } else {
-                    Command::new(program)
-                        .stdout(Stdio::null())
-                        .stderr(Stdio::null())
-                        .arg(path.as_ref())
-                        .spawn()?
-                        .wait()
-                }
-            })
+        let mut unsuccessful = None;
+        let mut error = None;
+
+        for (command, args) in &open_handlers {
+            let result = Command::new(command)
+                .stdout(Stdio::null())
+                .stderr(Stdio::null())
+                .args(*args)
+                .status();
+
+            match result {
+                Ok(status) if status.success() => return result,
+                Ok(_) => unsuccessful = unsuccessful.or(Some(result)),
+                Err(err) => error = error.or(Some(Err(err))),
+            }
+        }
+
+        unsuccessful
+            .or(error)
+            .unwrap_or_else(|| Err(io::Error::from(io::ErrorKind::Other)))
```

Comment:
```suggestion
            .expect("successful cases don't get here")
```

Is the above fair to say?

Reply:
`unwrap_or_else` should never be called. I was just afraid to panic here. The only way to reach the `unwrap_or_else` call would be a logic bug but it would be difficult to test for.